flax/linen/module.py

```diff
@@ -231,7 +228,7 @@ def wrapped_module_method(self, *args, **kwargs):
 
     if is_compact_method:
       if self.scope is None:
-        raise ValueError("Can't call compact methods on unbound modules")
+        raise UnboundCompactModuleError(self.name)
```

Comment:
I'm curious whether we also get this error when calling non-compact methods

Reply:
It doesn't seem to be possible since it states `if is_compact_pmethod` before, right?

Reply:
Right, I mean that it would be good to similarly detect that -- this was independently filed as #1014

Reply:
Good idea, I've moved your suggestion to the description of the new PR #1070 and I'll implement it there.